modules/dynamic/test/src-jvm/smithy4s/dynamic/ModelWrapper.scala

```diff
@@ -35,7 +35,7 @@ class ModelWrapper(val model: Model) {
   private final implicit class JavaStreamOps[A](
       stream: java.util.stream.Stream[A]
   ) {
-    def toList: List[A] = stream.collect(Collectors.toList()).asScala.toList
+    def asList: List[A] = stream.collect(Collectors.toList()).asScala.toList
```

Comment:
ok so I don't fully understand why it started to be a problem _now_ but it is indeed a problem for me.

I was getting "method `exists` doesn't exist on `java.util.List`" type errors. This is because in newer JDK versions there's a `toList()` method on `Stream` - it's been there since Java 16.

Thing is, we were already building on Java 17. I'm using Java 21 now as a default, so Metals probably has that as well, but my sbt in this project is still the one from the Nix flake, using Java 17.

I guess this could be a change in behavior between sbt or scala patch versions - didn't investigate more, but I find it slightly spooky that we can use newer Java features even though we have `-release 8` in our scalac options.

Reply:
I get it now. The `temurin-jre-bin-17` derivation from the Nix flake doesn't have `ct.sym` in it, so Scalac just uses all the available JDK methods instead of just the ones matching the target version. Strange choice but ok.